Global State readScopedState done in non blocking thread

### Describe the bug

While running some tests I found this failure intermittently due to calling a blocking call:

```
java.lang.AssertionError: Blocking call! java.io.FileInputStream#readBytes on thread Thread[#13191,non-blocking-thread-GracefulShutdownNetworkPartitionTest-NodeI-p92-t6,5,ISPN-non-blocking-group]
	at org.infinispan.util.CoreTestBlockHoundIntegration.lambda$applyTo$0(CoreTestBlockHoundIntegration.java:51)
	at reactor.blockhound.BlockHound$Builder.lambda$install$8(BlockHound.java:488)
	at reactor.blockhound.BlockHoundRuntime.checkBlocking(BlockHoundRuntime.java:89)
	at java.base/java.io.FileInputStream.readBytes(FileInputStream.java)
	at java.base/java.io.FileInputStream.read(FileInputStream.java:281)
	at java.base/sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:279)
	at java.base/sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:322)
	at java.base/sun.nio.cs.StreamDecoder.read(StreamDecoder.java:186)
	at java.base/java.io.InputStreamReader.read(InputStreamReader.java:183)
	at java.base/java.io.BufferedReader.fill(BufferedReader.java:166)
	at java.base/java.io.BufferedReader.readLine(BufferedReader.java:333)
	at java.base/java.io.BufferedReader.readLine(BufferedReader.java:400)
	at org.infinispan.globalstate.impl.GlobalStateHandler.readScopedState(GlobalStateHandler.java:83)
	at org.infinispan.globalstate.impl.GlobalStateManagerImpl.readScopedState(GlobalStateManagerImpl.java:131)
	at org.infinispan.topology.LocalTopologyManagerImpl.getMemberUUIDsFromState(LocalTopologyManagerImpl.java:352)
	at org.infinispan.topology.LocalTopologyManagerImpl.lambda$handleStatusRequest$0(LocalTopologyManagerImpl.java:506)
	at java.base/java.util.concurrent.CompletableFuture.uniApplyNow(CompletableFuture.java:705)
	at java.base/java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:683)
	at java.base/java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:2227)
	at java.base/java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:144)
	at org.infinispan.topology.LocalTopologyManagerImpl.handleStatusRequest(LocalTopologyManagerImpl.java:457)
	at org.infinispan.commands.topology.CacheStatusRequestCommand.invokeAsync(CacheStatusRequestCommand.java:36)
	at org.infinispan.topology.TopologyManagementHelper.invokeAsync(TopologyManagementHelper.java:151)
	at org.infinispan.topology.TopologyManagementHelper.executeOnClusterSync(TopologyManagementHelper.java:52)
	at org.infinispan.topology.ClusterTopologyManagerImpl.fetchClusterStatus(ClusterTopologyManagerImpl.java:653)
	at org.infinispan.topology.ClusterTopologyManagerImpl.recoverClusterStatus(ClusterTopologyManagerImpl.java:515)
	at org.infinispan.topology.ClusterTopologyManagerImpl.lambda$handleClusterView$0(ClusterTopologyManagerImpl.java:480)
	at org.infinispan.util.concurrent.ActionSequencer.safeNonBlockingCall(ActionSequencer.java:59)
	at org.infinispan.util.concurrent.ActionSequencer$SequenceEntry.run(ActionSequencer.java:228)
	at org.infinispan.util.concurrent.ActionSequencer$SequenceEntry.apply(ActionSequencer.java:221)
	at org.infinispan.util.concurrent.ActionSequencer$SequenceEntry.apply(ActionSequencer.java:181)
	at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:955)
	at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire$$$capture(CompletableFuture.java:932)
	at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java)
	at --- Async.Stack.Trace --- (captured by IntelliJ IDEA debugger)
	at java.base/java.util.concurrent.CompletableFuture$UniHandle.&lt;init&gt;(CompletableFuture.java:925)
	at java.base/java.util.concurrent.CompletableFuture.uniHandleStage(CompletableFuture.java:969)
	at java.base/java.util.concurrent.CompletableFuture.handleAsync(CompletableFuture.java:2409)
	at java.base/java.util.concurrent.CompletableFuture.handleAsync(CompletableFuture.java:144)
	at org.infinispan.util.concurrent.ActionSequencer$SequenceEntry.register(ActionSequencer.java:198)
	at org.infinispan.util.concurrent.ActionSequencer.registerAction(ActionSequencer.java:137)
	at org.infinispan.util.concurrent.ActionSequencer.orderOnKey(ActionSequencer.java:100)
	at org.infinispan.topology.ClusterTopologyManagerImpl.orderOnManager(ClusterTopologyManagerImpl.java:497)
	at org.infinispan.topology.ClusterTopologyManagerImpl.handleClusterView(ClusterTopologyManagerImpl.java:475)
	at org.infinispan.topology.ClusterTopologyManagerImpl.lambda$start$0(ClusterTopologyManagerImpl.java:175)
	at org.infinispan.topology.EventLoggerViewListener.handleViewChange(EventLoggerViewListener.java:39)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at org.infinispan.notifications.impl.AbstractListenerImpl$ListenerInvocationImpl.lambda$invoke$0(AbstractListenerImpl.java:431)
	at org.infinispan.notifications.impl.AbstractListenerImpl$ListenerInvocationImpl.invoke(AbstractListenerImpl.java:457)
	at org.infinispan.notifications.impl.AbstractListenerImpl.invokeListener(AbstractListenerImpl.java:358)
	at org.infinispan.notifications.impl.AbstractListenerImpl.invokeListeners(AbstractListenerImpl.java:348)
	at org.infinispan.notifications.cachemanagerlistener.CacheManagerNotifierImpl.notifyViewChange(CacheManagerNotifierImpl.java:102)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.receiveClusterView(JGroupsTransport.java:816)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport$ChannelCallbacks.up(JGroupsTransport.java:1638)
	at org.jgroups.JChannel.up(JChannel.java:669)
	at org.jgroups.stack.ProtocolStack.up(ProtocolStack.java:949)
	at org.jgroups.protocols.FRAG2.up(FRAG2.java:122)
	at org.jgroups.protocols.FlowControl.up(FlowControl.java:239)
	at org.jgroups.protocols.FlowControl.up(FlowControl.java:239)
	at org.jgroups.protocols.pbcast.GMS.installView(GMS.java:693)
	at org.jgroups.protocols.pbcast.ServerGmsImpl.handleViewChange(ServerGmsImpl.java:66)
	at org.jgroups.protocols.pbcast.GMS.castViewChangeAndSendJoinRsps(GMS.java:544)
	at org.jgroups.protocols.pbcast.CoordGmsImpl.handleMembershipChange(CoordGmsImpl.java:180)
	at org.jgroups.protocols.pbcast.GMS.process(GMS.java:1210)
	at org.jgroups.protocols.pbcast.ViewHandler.process(ViewHandler.java:239)
	at org.jgroups.protocols.pbcast.ViewHandler.add(ViewHandler.java:63)
	at org.jgroups.protocols.pbcast.GMS.handle(GMS.java:884)
	at org.jgroups.protocols.pbcast.GMS.up(GMS.java:797)
	at org.jgroups.protocols.pbcast.STABLE.up(STABLE.java:234)
	at org.jgroups.protocols.UNICAST3.deliverMessage(UNICAST3.java:1172)
	at org.jgroups.protocols.UNICAST3.addMessage(UNICAST3.java:898)
	at org.jgroups.protocols.UNICAST3.handleDataReceived(UNICAST3.java:880)
	at org.jgroups.protocols.UNICAST3.up(UNICAST3.java:490)
	at org.jgroups.protocols.pbcast.NAKACK2.up(NAKACK2.java:644)
	at org.jgroups.protocols.Discovery.up(Discovery.java:304)
	at org.jgroups.stack.Protocol.up(Protocol.java:365)
	at org.jgroups.protocols.DISCARD.up(DISCARD.java:167)
	at org.jgroups.protocols.TP.passMessageUp(TP.java:426)
	at org.jgroups.util.SubmitToThreadPool$SingleMessageHandler.run(SubmitToThreadPool.java:99)
	at java.base/java.util.concurrent.ThreadPerTaskExecutor$TaskRunner.run(ThreadPerTaskExecutor.java:291)
	at java.base/java.lang.VirtualThread.run(VirtualThread.java:460)
```

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

--- a/core/src/main/java/org/infinispan/topology/LocalTopologyManagerImpl.java
+++ b/core/src/main/java/org/infinispan/topology/LocalTopologyManagerImpl.java
@@ -454,7 +454,7 @@ public class LocalTopologyManagerImpl implements LocalTopologyManager, GlobalSta
    @Override
    public CompletionStage<ManagerStatusResponse> handleStatusRequest(int viewId) {
       // As long as we have an older view, we can still process topologies from the old coordinator
-      return withView(viewId, getGlobalTimeout(), MILLISECONDS).thenApply(ignored -> {
+      return blockingManager.thenApplyBlocking(withView(viewId, getGlobalTimeout(), MILLISECONDS), ignored -> {
          Map<String, CacheStatusResponse> caches = new HashMap<>();
          synchronized (runningCaches) {
             latestStatusResponseViewId = viewId;
@@ -509,7 +509,7 @@ public class LocalTopologyManagerImpl implements LocalTopologyManager, GlobalSta
 
          log.debugf("Sending cluster status response for view %d", viewId);
          return new ManagerStatusResponse(caches, gcr.getClusterTopologyManager().isRebalancingEnabled());
-      });
+      }, "LocalTopologyManager.handleStatusRequest-" + viewId);
    }
 
    @Override

